pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -456,10 +456,17 @@ protected void internalCreateNonPartitionedTopic(boolean authoritative) {
 
         validateTopicOwnership(topicName, authoritative);
 
-        PartitionedTopicMetadata partitionMetadata = getPartitionedTopicMetadata(topicName, authoritative, false);
-        if (partitionMetadata.partitions > 0) {
-            log.warn("[{}] Partitioned topic with the same name already exists {}", clientAppId(), topicName);
-            throw new RestException(Status.CONFLICT, "This topic already exists");
+        PartitionedTopicMetadata partitionMetadata = null;
+        try {
+            partitionMetadata = getPartitionedTopicMetadata(topicName, authoritative, false);
+            if (partitionMetadata.partitions > 0) {
+                log.warn("[{}] Partitioned topic with the same name already exists {}", clientAppId(), topicName);
+                throw new RestException(Status.CONFLICT, "This topic already exists");
+            }
+        } catch (RestException restException) {
+            if (Status.NOT_FOUND.getStatusCode() != restException.getResponse().getStatus()) {
+                throw restException;
```

Comment:
I think it's not correct handling here, If got 404 here, it means the partitioned metadata does not exist right? If the partitioned metadata does exist, why should we prevent the non-partitioned topic creation?